### ISSUE TITLE
[BUG] Export file names start with "mydumper_0" for databases with special characters in their name

### ISSUE THREAD

**Issue by sentia-be-ops** [mydumper/mydumper#748]
**Describe the bug**
When exporting databases with a special character in their name (e.g. '&'), the resulting file on disk starts with "mydumper_0" instead of the correct database name. This is not visible in the logs, where the database name is shown correctly with the special character.

When using myloader to import another database from the same backup, the files starting with "mydumper_0" get included as well (or used exclusively), causing import errors (duplicate keys, ...) and breaking the ability to import specific databases. This is visible in the logs as well.

**To Reproduce**
Command executed:
mydumper --less-locking --triggers --events --routines -o ~/mydumpertest/ -v 3 --use-savepoints --threads 2
myloader -d ~/mydumpertest -o -v 3 -s classicmodel -B restoretest

**Expected behavior**
Backup files should have the correct database name, including the special characters (as it was on older versions like 0.11.1 etc.)
Myloader should not include these "mydumper_0" files in imports of other, unrelated databases.

**Log**
```
mydumper --less-locking --triggers --events --routines -o ~/mydumpertest/ -v 3 --use-savepoints --threads 2
** Message: 12:06:52.585: MyDumper backup version: 0.12.3-3
** Message: 12:06:52.588: Server version reported as: 8.0.28-20
** Message: 12:06:52.588: Connected to a MySQL server
** Message: 12:06:52.589: Acquiring FTWRL
** Message: 12:06:52.595: Acquiring DDL lock
** Message: 12:06:52.596: Started dump at: 2022-07-07 12:06:52
** Message: 12:06:52.599: Thread 3 connected using MySQL connection ID 19
** Message: 12:06:52.600: Thread 4 connected using MySQL connection ID 20
** Message: 12:06:52.600: Thread 1 connected using MySQL connection ID 21
** Message: 12:06:52.601: Thread 2 connected using MySQL connection ID 22
** Message: 12:06:52.604: Thread 3 dumping db information for `ReferenceDB`
** Message: 12:06:52.610: Thread 4 dumping db information for `VD&S`
** Message: 12:06:52.613: Thread 3 dumping schema for `VD&S`.`voorraad_wwwfile`
** Message: 12:06:52.613: Shutdown jobs for less locking enqueued
** Message: 12:06:52.614: Waiting less locking jobs to complete
** Message: 12:06:52.615: Thread 3 dumping db information for `classicmodels`
** Message: 12:06:52.617: Thread 4 dumping db information for `mysql`
** Message: 12:06:52.646: Thread 3 dumping db information for `sys`
** Message: 12:06:52.682: Thread 4 shutting down
** Message: 12:06:52.682: Thread 2 dumping schema create for `VD&S`
** Message: 12:06:52.682: Thread 2 dumping data for `VD&S`.`voorraad_wwwfile` | Remaining jobs: 73
** Message: 12:06:52.684: Thread 2 dumping schema create for `mysql`
...
** Message: 12:06:52.686: Thread 2 dumping schema create for `classicmodels`
** Message: 12:06:52.687: Thread 2 dumping data for `classicmodels`.`customers` | Remaining jobs: 65
** Message: 12:06:52.689: Thread 2 dumping data for `classicmodels`.`employees` | Remaining jobs: 64
** Message: 12:06:52.691: Thread 2 dumping data for `classicmodels`.`offices` | Remaining jobs: 62
** Message: 12:06:52.693: Thread 2 dumping data for `classicmodels`.`orderdetails` | Remaining jobs: 60
** Message: 12:06:52.697: Thread 2 dumping data for `classicmodels`.`orders` | Remaining jobs: 60
** Message: 12:06:52.700: Thread 2 dumping data for `classicmodels`.`payments` | Remaining jobs: 59
** Message: 12:06:52.708: Thread 2 dumping data for `classicmodels`.`productlines` | Remaining jobs: 61
** Message: 12:06:52.709: Thread 2 dumping data for `classicmodels`.`products` | Remaining jobs: 59
** Message: 12:06:52.712: Thread 2 dumping schema create for `sys`
...
** Message: 12:06:52.795: Thread 3 shutting down
** Message: 12:06:52.795: Non-InnoDB dump complete, unlocking tables
** Message: 12:06:52.795: Thread 1 dumping view for `sys`.`schema_unused_indexes`
** Message: 12:06:52.796: Releasing DDL lock
** Message: 12:06:52.796: Shutdown jobs enqueued
** Message: 12:06:52.796: Waiting jobs to complete
...
** Message: 12:06:52.866: Thread 2 dumping SP and VIEWs for `sys`
** Message: 12:06:52.866: Thread 1 shutting down
** Message: 12:06:52.874: Thread 2 shutting down
** Message: 12:06:52.874: Releasing DDL lock
** Message: 12:06:52.875: Main connection closed
** Message: 12:06:52.875: Finished dump at: 2022-07-07 12:06:52
```

```
myloader -d ~/mydumpertest -o -v 3 -s classicmodel -B restoretest
** Message: 12:10:42.074: Server version reported as: 8.0.28-20
** Message: 12:10:42.078: DB post: restoretest for: sys-schema-post.sql
** Message: 12:10:42.078: Thread 2 restoring table `restoretest`.`voorraad_wwwfile` from //root/mydumpertest/mydumper_0.voorraad_wwwfile-schema.sql
** Message: 12:10:42.078: Dropping table or view (if exists) `restoretest`.`voorraad_wwwfile`
** Message: 12:10:42.079: Creating table `restoretest`.`voorraad_wwwfile` from content in //root/mydumpertest/mydumper_0.voorraad_wwwfile-schema.sql
** Message: 12:10:42.118: Thread 2 restoring `restoretest`.`voorraad_wwwfile` part 1 of 1 from mydumper_0.voorraad_wwwfile.00000.sql. Progress 1 of 1.
** Message: 12:10:42.124: Import timings:
** Message: 12:10:42.124: Data          | Index         | Total         | Table
** Message: 12:10:42.124: 0 00:00:00    | 0 00:00:00    | 0 00:00:00    | `restoretest`.`voorraad_wwwfile`
** Message: 12:10:42.124: Thread 3 restoring post on `restoretest` from sys-schema-post.sql
** Message: 12:10:42.431: Starting table checksum verification
```

**Backup**
```
ls -l | grep -v mysql | grep -v sys
total 2644
-rw-r--r-- 1 root root   16942 Jul  7 12:06 classicmodels.customers.00000.sql
-rw-r--r-- 1 root root    2139 Jul  7 12:06 classicmodels.employees.00000.sql
-rw-r--r-- 1 root root     791 Jul  7 12:06 classicmodels.offices.00000.sql
-rw-r--r-- 1 root root   95145 Jul  7 12:06 classicmodels.orderdetails.00000.sql
-rw-r--r-- 1 root root   28350 Jul  7 12:06 classicmodels.orders.00000.sql
-rw-r--r-- 1 root root   10991 Jul  7 12:06 classicmodels.payments.00000.sql
-rw-r--r-- 1 root root    3625 Jul  7 12:06 classicmodels.productlines.00000.sql
-rw-r--r-- 1 root root   30831 Jul  7 12:06 classicmodels.products.00000.sql
-rw-r--r-- 1 root root     134 Jul  7 12:06 classicmodels-schema-create.sql
-rw-r--r-- 1 root root      75 Jul  7 12:06 metadata
-rw-r--r-- 1 root root     153 Jul  7 12:06 mydumper_0-schema-create.sql
-rw-r--r-- 1 root root    6623 Jul  7 12:06 mydumper_0.voorraad_wwwfile.00000.sql
-rw-r--r-- 1 root root    1070 Jul  7 12:06 mydumper_0.voorraad_wwwfile-schema.sql
```

**How to repeat**
Create a database with a special character in its name, e.g. "VD&S".

**Environment (please complete the following information):**
 - OS version: Debian 11, percona MySQL 8.0.28
 - MyDumper version:
```
mydumper --version
mydumper 0.12.3-3, built against MySQL 8.0.28-19
```

```
myloader --version
myloader 0.12.3-3, built against MySQL 8.0.28-19
```

**Comment by sentia-be-ops** [comment]
@davidducos  I've tested with regex, but it doesn't seem to entirely work as expected as a workaround. I tested a few scenarios:

Restore a single DB without a regex, without changing the name of the DB to restore to (as expected this includes unrelated data/databases which is quite dangerous!):
`myloader -d ~/mydumpertest -o -v 3 -s classicmodels --logfile noregex-DB-to-DB.log`
[noregex-DB-to-DB.log](https://github.com/mydumper/mydumper/files/9064108/noregex-DB-to-DB.log)

Restore a single DB with a regex, without changing the name of the DB to restore to (this seems to work as expected):
`myloader -d ~/mydumpertest -o -v 3 --regex '^(classicmodels)' -s classicmodels --logfile regex-DB-to-DB.log`
[regex-DB-to-DB.log](https://github.com/mydumper/mydumper/files/9064088/regex-DB-to-DB.log)

Restore a single DB without regex, changing the target DB (as expected this includes unrelated data/databases, but less dangerous than scenario 1):
`myloader -d ~/mydumpertest -o -v 3 -s classicmodels -B restoretest --logfile noregex-DB-to-other-DB.log`
[noregex-DB-to-other-DB.log](https://github.com/mydumper/mydumper/files/9064145/noregex-DB-to-other-DB.log)

Restore a single DB with an regex, changing the target DB (doesn't work as expected, skips everything due to the database name change):
`myloader -d ~/mydumpertest -o -v 3 -s classicmodels --regex '^(classicmodels)' -B restoretest --logfile regex-DB-to-other-DB.log`
[regex-DB-to-other-DB.log](https://github.com/mydumper/mydumper/files/9064161/regex-DB-to-other-DB.log)

So restoring to the original database seems to work, but I've not found a working regex to exclude the unrelated data/databases when restoring to another destination database. Any suggestions?
